pkg/cmd/gist/shared/shared.go

```diff
@@ -67,3 +68,26 @@ func GistIDFromURL(gistURL string) (string, error) {
 
 	return "", fmt.Errorf("Invalid gist URL %s", u)
 }
+
+// DetectedBinaryFileContents detects if a file is binary.
+// isPhysicalFile is `true` only when `file` is a path to an actual physical file.
+// If `file` is a byte stream or string content, isPhysicalFile is `false`.
+func DetectedBinaryFileContents(file string, isPhysicalFile bool) (bool, error) {
```

Comment:
This function feels like it wants to be two separate functions. The fact that the 2nd argument defines the type of the 1st argument feels like a code smell to me.

One function should be for checking physical file contents. It should be called `IsBinaryFile` and take a string filename to read from disk.

Another function should be called `IsBinaryContents` and it should take `[]byte` input.

It's fine if there will be a bit of duplication between the two functions; specifically the `mime.Parent()` for loop will be duplicated.

Reply:
Actually on a second thought, even `IsBinaryFile` isn't needed because while creating a gist, we read the contents of the file and then passed it on to mimetype to remove a redundant file operation remember? We only need `IsBinaryContents` on the whole. What do you say? @mislav